torch/utils/_python_dispatch.py

```diff
@@ -164,3 +164,130 @@ def transform_subclass(t, callback):
+        shared_aliases = arg_aliases & out_aliases
+        return len(shared_aliases) > 0 and not arg.alias_info.is_write
+
+    num_args = len(func._schema.arguments)
+    num_returns = len(func._schema.returns)
+    for arg_idx in range(num_args):
+        for return_idx in range(num_returns):
+            if is_read_only_alias_match(func._schema.arguments[arg_idx], func._schema.returns[return_idx]):
+                alias_storage(args[arg_idx], outs[return_idx])
+
+    # Sigh... the torchscript parser has a bug where alias annotations for Tensor[](a) don't show up properly
+    # See https://github.com/pytorch/pytorch/issues/106173
+    if func.overloadpacket in [
+        torch.ops.aten.chunk,
+        torch.ops.aten.tensor_split,
+        torch.ops.aten.split,
+        torch.ops.aten.split_with_sizes,
+        torch.ops.aten.hsplit,
+        torch.ops.aten.vsplit,
+        torch.ops.aten.dsplit,
+        torch.ops.aten.unbind,
+    ]:
+        assert isinstance(outs, list) and all(isinstance(x, torch.Tensor) for x in outs)
+        for o in outs:
+            # For lists of outputs, need to alias every individual tensor to the input
+            alias_storage(args[0], o)
+
+def return_and_correct_aliasing(func, args, kwargs, out):
+    def get_write_alias(x):
+        if not x.alias_info or not x.alias_info.before_set:
+            return None
+        before_set = list(x.alias_info.before_set)
+        # torchscript allows for complicated alias sets, but our dispatcher ops only really involve simple aliasing
+        assert len(before_set) == 1
+        if x.alias_info.is_write:
+            return before_set[0]
+        return None
+
+    def get_arg_idx_from_alias(output_alias):
+        arg_indices = [
+            i for i, a in enumerate(func._schema.arguments)
+            if a.alias_info is not None and output_alias in a.alias_info.before_set
+        ]
+        # For any dispatcher op with an output alias, we expect it to map to exactly one alias in the schema's input arguments.
+        assert len(arg_indices) == 1
+        return arg_indices[0]
+
+    # Fix up the storages of any outs so that they point to the same storage as the input,
+    # if func is a view op.
+    _correct_storage_aliasing(func, args, [out] if not isinstance(out, (list, tuple)) else out)
+
+    # For inplace_view ops in particular, we'll try hard to make sure that the wrapper subclass's
+    # metadata is set correctly.
+    if torch.Tag.inplace_view in func.tags:
+        # no_dispatch() to make sure that we secretly change the metadata on the wrapper,
+        # but don't end up dispatching the op anywhere else.
+        mutated_args = [x for i, x in enumerate(args) if get_write_alias(func._schema.arguments[i]) is not None]
+        # Assumption: we have a very small number of inplace_view ops that follow a strict schema:
+        # there is only a single argument that gets its metadata mutated.
+        assert len(mutated_args) == 1
```

Comment:
Is it actually a problem if you have more than one mutated arg? I don't see what would fail even if there were multiple

Reply:
Yeah it shouldn't be a problem - so far I wrote this function in a more limited way that doesn't handle arbitrary schemas, and instead only handles the subset of schemas that I know exist in aten (for example, inplace_view ops always have a single mutated argument).

Happy to improve it as part of this PR, I just figured I wouldn't bother complicating it until someone complains.